### PR TITLE
Add "warmth" style const with demo

### LINKED ISSUE #306: Add temperature and warmth scales to style class demo
What it says on the tin.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/77799349-add-temperature-and-warmth-scales-to-style-class-demo?utm_campaign=plugin&utm_content=tracker%2F60830613&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60830613&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by dar5hak**
Relates to #306 ~(or not; I don't know what the "temperature" class referred there is)~.

Here's what the demo looks like:

Screenshot from 2020-08-24 22-49-07

Screenshot from 2020-08-24 22-50-29

### CODE REVIEW

**Comment by dar5hak** [comment]
Once this is released in a version, we can update the [Night Light code](https://github.com/elementary/switchboard-plug-display/blob/master/src/Views/NightLightView.vala#L80).

File: demo/Views/CSSView.vala [hunk 1]
```diff
@@ -84,6 +84,15 @@ public class CSSView : Gtk.Grid {
         back_button.halign = Gtk.Align.START;
         back_button.get_style_context ().add_class (Granite.STYLE_CLASS_BACK_BUTTON);
 
+        var warmth_label = new Gtk.Label ("\"warmth\" style class:");
+        warmth_label.halign = Gtk.Align.END;
```
Comment (cassidyjames) [inline demo/Views/CSSView.vala:88]: Modern code style

```suggestion
        var warmth_label = new Gtk.Label ("\"warmth\" style class:") {
            halign = Gtk.Align.END
        };
```

File: demo/Views/CSSView.vala [hunk 2]
```diff
@@ -84,6 +84,15 @@ public class CSSView : Gtk.Grid {
...
+
+        var warmth_scale = new Gtk.Scale.with_range (Gtk.Orientation.HORIZONTAL, 3500, 6000, 10);
+        warmth_scale.draw_value = false;
+        warmth_scale.has_origin = false;
+        warmth_scale.inverted = true;
+        warmth_scale.get_style_context ().add_class (Granite.STYLE_CLASS_WARMTH);
```
Comment (cassidyjames) [inline demo/Views/CSSView.vala:94]: Modern code style, plus horizontal expand

```suggestion
        var warmth_scale = new Gtk.Scale.with_range (Gtk.Orientation.HORIZONTAL, 3500, 6000, 10) {
            draw_value = false,
            has_origin = false,
            hexpand = true,
            inverted = true
        };
        warmth_scale.get_style_context ().add_class (Granite.STYLE_CLASS_WARMTH);
```

**Comment by cassidyjames** [comment]
@dar5hak the `temperature` slider is another one we use in Photos. Literally just `temperature`. It can be a similar but follow-up branch. :)

**Comment by dar5hak** [comment]
Ah, so that's what `temperature` meant. All this time, I was wondering where in the entire OS we deal with thermodynamic temperature. 😅

I'll try and add it in the evening (that's Tuesday morning in the US).

**Comment by dar5hak** [comment]
@cassidyjames I've changed the "fixes" comment so it won't automatically close the issue. Can you maybe merge this so that the temperature branch doesn't conflict for the `Gtk.Grid` indices?

**Comment by dar5hak** [comment]
BTW @cassidyjames thanks for the tip on modern style for object property assignment. I went and updated the same in Badger as well. 😀🙌

**Comment by cassidyjames** [comment]
@dar5hak awesome! Yeah that's a pretty recent thing we aligned on code-style-wise. It feels a lot nicer. Thanks for contributing this!

**Pull request merged**